File: output/ChatGPT/avatar/Java/Python/codeforces_607_A.py
n = int(input())
beacon = [0] * 1000001
dp = [0] * 1000001

for i in range(n):
    a, b = map(int, input().split())
    beacon[a] = b

if beacon[0] != 0:
    dp[0] = 1

for i in range(1, 1000001):
    if beacon[i] != 0 and beacon[i] < i:
        dp[i] = dp[i - beacon[i] - 1] + 1
    elif beacon[i] != 0:
        dp[i] = 1
    else:
        dp[i] = dp[i - 1]

max_val = max(dp)
print(n - max_val)

File: output/ChatGPT/avatar/Java/C++/codeforces_607_A.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int max = 0;
    int n;
    cin >> n;
    vector<int> beacon(1000001, 0);
    vector<int> dp(1000001, 0);

    for (int i = 0; i < n; i++) {
        int a, b;
        cin >> a >> b;
        beacon[a] = b;
    }

    if (beacon[0] != 0) dp[0] = 1;

    for (int i = 1; i < 1000001; i++) {
        if (beacon[i] != 0 && beacon[i] < i) {
            dp[i] = dp[i - beacon[i] - 1] + 1;
        } else if (beacon[i] != 0) {
            dp[i] = 1;
        } else {
            dp[i] = dp[i - 1];
        }
        max = max(max, dp[i]);
    }

    cout << n - max << endl;

    return 0;
}